### ISSUE TITLE
Setup IP tables rules to redirect inbound/outbound traffic to Envoy sidecar

### ISSUE THREAD

**Issue by rshriram** [amalgam8/amalgam8#482]
Traffic from the app can be transparently forwarded to the Envoy sidecar, without the application needing to explicitly talk to Envoy.

The following IP tables rules need to be used (as done in Istio):

```bash
sudo iptables -t nat -A PREROUTING -p tcp -j REDIRECT --to-port 5001
sudo iptables -t nat -A OUTPUT -p tcp -j REDIRECT --to-port 5001 -m owner '!' --uid-owner 3333 sudo iptables -t nat --list
```

In order to run this rule, a new user with id 3333 needs to be created before launching Envoy. Envoy then needs to be run under this user id, so that packets from Envoy don't get redirected to itself.

How it works:
 Apps can talk to each other without being aware of the sidecar. They can address each other with DNS names or even IP addresses. Traffic gets redirected to Envoy automatically on port 5001. Envoy sidecar supports SO_DST_PORT option and thus identifies the src_port to which responses should be forwarded back. Envoy forwards the request upstream, gets the response and uses the stored port to send the response back to the right caller.

The IP tables rules can be run in any docker container as it has its own network namespace (k8s and docker swarm). 

- For Docker swarm/compose, we need to run this as a shell script before starting the sidecar manager and envoy. This script needs to be packaged as part of sidecar and needs to be run by the sidecar manager.
- For Kubernetes, users can customize the init container to set this up before the sidecar container is started. See this example from istio/manager (https://github.com/istio/manager/blob/master/test/example/echo/echo.yaml#L28)

**Comment by jemcginn** [comment]
P4 - low priority
